lib/Backends/OpenCL/OpenCLDeviceManager.h

```diff
@@ -35,20 +35,39 @@ namespace runtime {
 class OpenCLBuffer {
   /// The OpenCL buffer being stored.
   cl_mem buffer_;
+
   /// Count of functions using this buffer.
   unsigned int users_{0};
+
   /// Size of the buffer in bytes.
   const size_t size_{0};
 
+  /// CL command queue for the network.
+  cl_command_queue commands_;
```

Comment:
Why do you associate a `cl_command_queue` with `OpenCLBuffer`? It is kind of strange. In theory, the same buffer could be used by different command queues.

Reply:
I think a rename might be good here, the idea being there is one command queue for all functions that share a set of common weights since no more than one can run at a time. Alternatively I could create a command queue for each function.